src/generate.jl

```diff
@@ -52,38 +52,40 @@ generator.
 function emit! end
 
 # TODO: Probably should be dropped but are nice as they reduce churn while iterating on design
-num_rows(g::TableGenerator, state::Nothing) = num_rows(g)
-emit!(g::TableGenerator, deps, state::Nothing) = emit!(g, deps)
+num_rows(rng, g::TableGenerator, state::Nothing) = num_rows(rng, g)
+emit!(rng, g::TableGenerator, deps, state::Nothing) = emit!(rng, g, deps)
 
-function generate(dag; size::Integer=10)
+generate(dag; kwargs...) = generate(Random.GLOBAL_RNG, dag; kwargs...)
+
+function generate(rng::AbstractRNG, dag; size::Integer=10)
     channel = Channel(size) do ch
-        return generate(dag) do table, row
+        return generate(rng, dag) do table, row
             return put!(ch, table => row)
         end
     end
 
     return channel
 end
 
-function generate(callback, dag)
-    return _generate!(callback, dag, Dict())
+function generate(callback, rng::AbstractRNG, dag)
+    return _generate!(rng, callback, dag, Dict())
 end
 
-function _generate!(callback, dag::AbstractVector, deps)
+function _generate!(rng::AbstractRNG, callback, dag::AbstractVector, deps)
```

Comment:
Keep the order consistent with `generate`:
```suggestion
function _generate!(callback, rng::AbstractRNG, dag::AbstractVector, deps)
```